### PR TITLE
RPC method to count the number of TX in the host

### PR DESCRIPTION
**Pull request by badgersrus**
### Why this change is needed

Moray needs the true value of the tx count for the e2e tests so we can check for overflow. 
https://github.com/ten-protocol/ten-internal/issues/3377

### What changes were made as part of this PR

* New RPC-only method to run a count over the tx table

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/host/storage/interfaces.go [hunk 1]
```diff
@@ -44,6 +44,8 @@ type BatchResolver interface {
 	FetchBatchByHeight(height *big.Int) (*common.PublicBatch, error)
 	// FetchTotalTxCount returns the number of transactions in the DB
 	FetchTotalTxCount() (*big.Int, error)
+	// FetchTotalTxsQuery returns the number of transactions in the DB
```
Comment (BedrockSquirrel) [inline go/host/storage/interfaces.go:47]: Need to give this a different comment to the one above imo, clarify why these two identical looking methods exist a bit haha.

File: go/host/storage/interfaces.go [hunk 2]
```diff
@@ -44,6 +44,8 @@ type BatchResolver interface {
 	FetchBatchByHeight(height *big.Int) (*common.PublicBatch, error)
 	// FetchTotalTxCount returns the number of transactions in the DB
```
Comment (tudor-malene) [inline go/host/storage/interfaces.go:45]: maybe explain in the comment the purpose of these 2 methods.
A fast but inaccurate one to be used for tenscan, and the other for testing

**Review by tudor-malene — APPROVED** [review]
lgtm. 
(one small comment to add)

**Review by BedrockSquirrel — APPROVED** [review]
LGTM - except maybe distinguish a bit why the two methods are different (I guess the gist is you want a cached value because worried that row count is expensive but also want it to be ticking up for optics).

Personally I'm not sure the tx count should be persisted, if it's not been set in the last half hour or something it should query the count, and then in memory just count it up as batch txs come in. But this works for now so whatever's easier.
